diff --git a/libpiper/include/piper_impl.hpp b/libpiper/include/piper_impl.hpp
index aa12a1d..186d297 100644
--- a/libpiper/include/piper_impl.hpp
+++ b/libpiper/include/piper_impl.hpp
@@ -35,7 +35,7 @@ const float DEFAULT_NOISE_W_SCALE = 0.8f;
 const int DEFAULT_HOP_LENGTH = 256;

 // onnx
-Ort::Env ort_env{ORT_LOGGING_LEVEL_WARNING, "piper"};
+extern Ort::Env ort_env;

 // espeak
 #define CLAUSE_INTONATION_FULL_STOP 0x00000000
@@ -87,7 +87,7 @@ struct piper_synthesizer {
 };

 // Get the first UTF-8 codepoint of a string
-std::optional<Phoneme> get_codepoint(std::string s) {
+inline std::optional<Phoneme> get_codepoint(std::string s) {
     auto view = una::views::utf8(s);
     auto it = view.begin();

diff --git a/libpiper/src/piper.cpp b/libpiper/src/piper.cpp
index 774e87d..208a71e 100644
--- a/libpiper/src/piper.cpp
+++ b/libpiper/src/piper.cpp
@@ -9,6 +9,8 @@

 using json = nlohmann::json;

+Ort::Env ort_env{ORT_LOGGING_LEVEL_WARNING, "piper"};
+
 struct piper_synthesizer *piper_create(const char *model_path,
                                        const char *config_path,
                                        const char *espeak_data_path) {
